Rakefile

```diff
@@ -15,3 +16,49 @@ namespace :plugins do
     t.pattern = "plugins/*/test/**/*_test.rb"
   end
 end
+
+namespace :test do
+  task js: :environment do
+    with_tmp_karma_config do |config|
+      sh "./node_modules/karma/bin/karma start #{config} --single-run"
+    end
+  end
+
+  private
+
+  def with_tmp_karma_config
+    Tempfile.open('karma.js') do |f|
+      f.write build_karma_config(karma_test_files)
+      f.flush
+      yield f.path
+    end
+  end
+
+  def karma_test_files
+    [
+      resolve_asset('vis.js'),
+      'vendor/assets/javascripts/angular.min.js',
+      'vendor/assets/javascripts/angular-mocks.js',
+      'vendor/assets/javascripts/underscore.min.js',
+      'test/angular/test_helper.js',
+      'app/assets/javascripts/app.js',
+      'app/assets/javascripts/controllers/**/*.js',
+      'app/assets/javascripts/directives/**/*.js',
+      'app/assets/javascripts/timeline.js',
+      'test/angular/**/*_spec.js'
+    ]
+  end
+
+  def resolve_asset(file)
+    Rails.application.assets.find_asset(file).to_a.first.pathname.to_s
+  end
+
+  def build_karma_config(files)
+    config = File.read('test/karma.conf.js')
+    {
+      "INSERT_FILES_VIA_RAKE" => "\"#{files.join("\",\n\"")}\"",
+      "INSERT_BASE_PATH_VIA_RAKE" => Bundler.root.to_s,
+    }.each { |key, value| config.gsub!(key, value) || raise("#{key} not found") }
```

Comment:
Could we not do all this via ERB and keep this task clean with a few ERB directives in the karma.conf.js file?